specification/batch/resource-manager/Microsoft.Batch/stable/2021-06-01/BatchManagement.json

```diff
@@ -0,0 +1,5878 @@
+          "title": "The URI of the file system to mount."
+        },
+        "relativeMountPath": {
+          "type": "string",
+          "title": "The relative path on the compute node where the file system will be mounted",
+          "description": "All file systems are mounted relative to the Batch mounts directory, accessible via the AZ_BATCH_NODE_MOUNTS_DIR environment variable."
+        },
+        "mountOptions": {
+          "type": "string",
+          "title": "Additional command line options to pass to the mount command.",
+          "description": "These are 'net use' options in Windows and 'mount' options in Linux."
+        },
+        "password": {
+          "type": "string",
+          "title": "The password to use for authentication against the CIFS file system."
+        }
+      },
+      "required": [
+        "username",
+        "source",
+        "password",
+        "relativeMountPath"
+      ],
+      "title": "Information used to connect to a CIFS file system."
+    },
+    "AzureFileShareConfiguration": {
+      "type": "object",
+      "properties": {
+        "accountName": {
+          "type": "string",
+          "title": "The Azure Storage account name."
+        },
+        "azureFileUrl": {
+          "type": "string",
+          "title": "The Azure Files URL.",
+          "description": "This is of the form 'https://{account}.file.core.windows.net/'."
+        },
+        "accountKey": {
+          "type": "string",
+          "title": "The Azure Storage account key."
+        },
+        "relativeMountPath": {
+          "type": "string",
+          "title": "The relative path on the compute node where the file system will be mounted",
+          "description": "All file systems are mounted relative to the Batch mounts directory, accessible via the AZ_BATCH_NODE_MOUNTS_DIR environment variable."
+        },
+        "mountOptions": {
+          "type": "string",
+          "title": "Additional command line options to pass to the mount command.",
+          "description": "These are 'net use' options in Windows and 'mount' options in Linux."
+        }
+      },
+      "required": [
+        "accountName",
+        "azureFileUrl",
+        "accountKey",
+        "relativeMountPath"
+      ],
+      "title": "Information used to connect to an Azure Fileshare."
+    },
+    "OutboundEnvironmentEndpointCollection": {
+      "type": "object",
+      "properties": {
+        "value": {
+          "type": "array",
+          "readOnly": true,
+          "items": {
+            "$ref": "#/definitions/OutboundEnvironmentEndpoint"
+          },
+          "description": "The collection of outbound network dependency endpoints returned by the listing operation."
+        },
+        "nextLink": {
+          "type": "string",
+          "description": "The continuation token."
+        }
+      },
+      "description": "Values returned by the List operation."
+    },
+    "OutboundEnvironmentEndpoint": {
```

Comment:
This resource is not defined as per ARM contract.
Any ARM resource should contain id/name/type as top level properties and RP specific properties should be defined under properties envelope.

Refer https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/resource-api-reference.md#resource-request-properties-envelope

Reply:
OutboundNetworkDependenciesEndpoints are not a true customer-owned ARM resource and do not have ids, names, tags etc. This endpoint gives some read-only information about batchAccounts and does not represent a sub-resource of batchAccounts.

This endpoint is a widespread Vanguard requirement made according to a Networking spec. It should be noted that many teams are implementing endpoints exactly like this one and some teams have already gotten them approved+merged+released.
See AKS at https://github.com/Azure/azure-rest-api-specs/pull/14147/files and a similar dispute at https://github.com/Azure/azure-rest-api-specs/pull/14332#discussion_r629606824

Reply:
Alright, sounds good. Signing off as the same pattern exists across other services.